pkg/k8s/watchers/cilium_endpoint_batch.go

```diff
@@ -0,0 +1,84 @@
+//  Copyright 2021 Authors of Cilium
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package watchers
+
+import (
+	"sync"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	cilium_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/k8s/informer"
+	"github.com/cilium/cilium/pkg/k8s/watchers/subscriber"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	cebSubscriber = subscriber.NewRaw()
+)
+
+func (k *K8sWatcher) ciliumEndpointBatchInit(client *k8s.K8sCiliumClient, asyncControllers *sync.WaitGroup) {
+	log.Info("Initializing CEB controller")
+	var once sync.Once
+
+	// Register for all ceb updates.
+	cebSubscriber.Register(newCEBSubscriber(k))
```

Comment:
Rather than dep inject the entire `K8sWatcher`, let's instead only inject individual isolated deps. Otherwise, the purpose of the refactor is rendered moot.

Reply:
On second thought, breaking up the `K8sWatcher` dependency will depend on converting the CEP handling into a separate subscriber[1] like you've done here for CEB. If done, then this refactor should be done in its own commit for the sake of bisect. You can either do this refactor now as part of this PR, or do it as a followup as another PR into the long-running branch. Either way it will need to be done because we want all new additions to the K8s watchers to be using the new scheme and over time, all other watcher will get converted.

Reply:
`CEB` relies on the `CEP` dependencies. Breaking CEP dependencies should be handled in another PR, not in this PR.

Reply:
Tracking the request change in GH [Issue](https://github.com/cilium/cilium/issues/17355)